### DISCUSSION TITLE
Specify/Match TLS cipher with network request

### DISCUSSION THREAD

**Discussion by carlin-q-scott** [projectdiscovery/nuclei#1669] [Ideas]
I used [this script for NMAP](https://nmap.org/nsedoc/scripts/ssl-enum-ciphers.html) to scan my site for available ciphers. It would be nice if I could run it with nuclei and match known weak/vulnerable TSL ciphers.

**Comment by ehsandeep** [comment]
Hey @carlin-q-scott, thanks for sharing this, nuclei engine have dedicated module for SSL that supports feeding custom tls connection with custom ciphers as input, here is an example [template]( https://github.com/projectdiscovery/nuclei-templates/blob/master/ssl/deprecated-tls.yaml), unfortunately docs are not in [sync/updated](https://github.com/projectdiscovery/nuclei/issues/1657) and we still have to detail [all the information](https://github.com/projectdiscovery/nuclei/pull/1529) to write more complex SSL templates, will look further into shared NSE script to explore possibilities of new improvement.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
